contracts/crowdsale/Crowdsale.sol

```diff
@@ -73,7 +78,8 @@ contract Crowdsale {
     uint256 weiAmount = msg.value;
 
     // calculate token amount to be created
-    uint256 tokens = weiAmount.mul(rate);
+    uint256 _rate = getRate();
+    uint256 tokens = weiAmount.mul(_rate);
```

Comment:
Wouldn't extracting `getTokenAmount` with default implementation of `weiAmount.mul(rate)` be more flexible?

Reply:
I am not really sure which one is more flexible :) I wanted to keep the code as similar to the original one

Reply:
I had two scenarios in my mind:
1) Applying multiple price levels to a transaction, for example:
    - You have a crowdsale with a bonus period where you are minting tokens at 25% increased rate until GOAL wei are raised.
   - Let's say you have raised (GOAL - 10 wei) already and a big transaction comes sending 10k ETH. You only want to mint (10 wei * 25%) worth of bonus tokens (not 10K ETH * 25 %).
2) Rounding problems when original rate = 1. Example:
   -  If you want to implement 1 ETH = 1 Token crowdsale, you'll use a rate of 1 and a token with 18 decimal places.
   - Now if you want to issue tokens at 125% rate at first, you have a problem, as `getRate` result will be `1 (original rate) * 5/4`, which rounds down to `1`, hence, no bonus.

It is impossible to implement any of these two by just overriding `getRate()`.

As you can always override `getTokenAmount(weiAmount)` and implement it as `weiAmount.mul(getRate())` with a custom `getRate()` function, to me it seems like a superior approach.

Reply:
You are right, will rewrite it ASAP ;)